packages/state-transition/src/cache/stateCache.ts

```diff
@@ -176,22 +176,22 @@ export function loadCachedBeaconState<T extends BeaconStateAllForks & BeaconStat
     stateBytes,
     seedValidatorsBytes
   );
-  const {pubkey2index, index2pubkey} = cachedSeedState.epochCtx;
+  const {finalizedPubkey2index, finalizedIndex2pubkey} = cachedSeedState.epochCtx;
   // Get the validators sub tree once for all the loop
   const validators = migratedState.validators;
   for (const validatorIndex of modifiedValidators) {
     const validator = validators.getReadonly(validatorIndex);
     const pubkey = validator.pubkey;
-    pubkey2index.set(pubkey, validatorIndex);
-    index2pubkey[validatorIndex] = bls.PublicKey.fromBytes(pubkey, CoordType.jacobian);
+    finalizedPubkey2index.set(pubkey, validatorIndex);
```

Comment:
should not update finalized validator data here as they are shared across states, update `unfinalizedPubkey2index` instead